### ISSUE TITLE
Incorrect block reward values since 2023-07-26

### ISSUE THREAD

**Issue by AlexLloyd0** [mempool/mempool#4073]
### Description

Since 26th July we've been observing some incorrect results returned by the mempool.space API.

### Steps to reproduce

This bug does not reliably reproduce, making the same request now returns the correct value. I have logs of requests and responses, listed below. Each of these came from an `/api/v1/blocks/<height>` request, e.g. https://mempool.space/api/v1/blocks/800513. We get the reward from `extras.reward`

This code has been running for 3 months without issue until last week (and doesn't coincide with an update on our end).

### Expected behaviour

Return the correct block reward.

### Actual behaviour

Return a number that can be multiple orders of magnitude off.

#### Additional info

Request time (+- a few seconds) | height | block.extras.reward | block.extras.totalFees 
--------------------------- | -------- | ---------------- | -------------------
2023-07-31 02:05:10.548486 | 800972 |       50578497 |           9975968
2023-07-31 02:05:10.548486 | 800965 |    72512601878 |          17078021
2023-07-31 02:05:10.548486 | 800964 |      136956686 |           7161294
2023-07-31 02:05:10.548486 | 800950 |    51617804846 |           8983713
2023-07-31 02:05:10.548486 | 800885 |      131383800 |           6707144
2023-07-31 02:05:10.548486 | 800862 |     2539587898 |           7912585
2023-07-30 02:05:04.617109 | 800762 |    43444657955 |          15635616
2023-07-29 02:05:14.927024 | 800660 |        5933108 |          23562653
2023-07-29 02:05:03.65396  | 800625 |      314746025 |           7839634
2023-07-29 02:05:11.688367 | 800568 |    36736928441 |           9196698
2023-07-28 02:05:02.569148 | 800545 |   195005625790 |          19522170
2023-07-27 02:05:02.762216 | 800409 |    26452911389 |          17475799
2023-07-27 02:05:08.80199  | 800311 |        2475569 |           7236792
2023-07-27 02:05:08.80199  | 800285 |     1266018269 |          11588357
2023-07-26 02:05:08.788329 | 800242 |       18297044 |          24846456

**Comment by wiz** [comment]
Yes, apologizes for this, apparently we had a bug which caused some inconsistently incorrect data to be returned starting from a few thousand blocks ago. We are aware of the problem and working to correct it ASAP.

**Comment by mononaut** [comment]
should be fixed by #4050

**Comment by AlexLloyd0** [comment]
Thanks for the fix!

Just confirming - the PR only mentions missing transactions, but some block rewards are reported as orders of magnitude too large, which I wouldn't expect to be caused by _missing_ transactions. I don't have the context though so if you say it's a fix I'll be satisfied.

Are you able to give an estimate of when this will hit prod?
